package.json

```diff
@@ -20,7 +20,8 @@
     "solium": "solium --dir ./contracts/",
     "truffle": "truffle",
     "build": "rimraf build && truffle compile",
-    "docs:gen": "node docs/gen/gen.js"
+    "docs:gen": "node docs/scripts/gen.js",
+    "publish": "npm run docs:gen"
```

Comment:
publish should build truffle + get docs + publish npm .
Something like : 
"truffle build && npm run docs:gen && network --clean && npm publish"

Reply:
sorry, thought it was automatic.